formats/config/section.go

```diff
@@ -0,0 +1,76 @@
+package config
+
+import "strings"
+
+type Section struct {
+	Name        string
+	Options     Options
+	Subsections Subsections
+}
+
+type Subsection struct {
+	Name    string
+	Options Options
+}
+
+type Sections []*Section
```

Comment:
`Options`, `Sections` and `Subsections` are currently slices. Willl it make sense to use maps instead, given that we are somewhat assuming names will be unique ids?